node/src/accountant/mod.rs

```diff
@@ -1411,131 +886,129 @@ mod tests {
+            .make_result(PayableDaoMock::new()) // For Payable Scanner
+            .make_result(PayableDaoMock::new()); // For PendingPayable Scanner
+        let pending_payable_dao_factory = PendingPayableDaoFactoryMock::new()
+            .make_params(&pending_payable_dao_factory_params_arc)
+            .make_result(PendingPayableDaoMock::new()) // For Accountant
+            .make_result(PendingPayableDaoMock::new()) // For Payable Scanner
+            .make_result(PendingPayableDaoMock::new()); // For PendingPayable Scanner
+        let receivable_dao_factory = ReceivableDaoFactoryMock::new()
+            .make_params(&receivable_dao_factory_params_arc)
+            .make_result(ReceivableDaoMock::new()) // For Accountant
+            .make_result(ReceivableDaoMock::new()); // For Receivable Scanner
+        let banned_dao_factory = BannedDaoFactoryMock::new()
+            .make_params(&banned_dao_factory_params_arc)
+            .make_result(BannedDaoMock::new()); // For Receivable Scanner
 
         let _ = Accountant::new(
-            &config,
+            &mut config,
             Box::new(payable_dao_factory),
             Box::new(receivable_dao_factory),
             Box::new(pending_payable_dao_factory),
             Box::new(banned_dao_factory),
         );
 
-        assert_eq!(payable_dao_factory_called.as_ref(), &RefCell::new(true));
-        assert_eq!(receivable_dao_factory_called.as_ref(), &RefCell::new(true));
         assert_eq!(
-            pending_payable_dao_factory_called.as_ref(),
-            &RefCell::new(true)
+            *payable_dao_factory_params_arc.lock().unwrap(),
+            vec![(), (), ()]
+        );
+        assert_eq!(
+            *pending_payable_dao_factory_params_arc.lock().unwrap(),
+            vec![(), (), ()]
+        );
+        assert_eq!(
+            *receivable_dao_factory_params_arc.lock().unwrap(),
+            vec![(), ()]
         );
-        assert_eq!(banned_dao_factory_called.as_ref(), &RefCell::new(true));
+        assert_eq!(*banned_dao_factory_params_arc.lock().unwrap(), vec![()]);
     }
 
     #[test]
     fn accountant_have_proper_defaulted_values() {
-        let mut bootstrapper_config = BootstrapperConfig::new();
-        bootstrapper_config.accountant_config_opt =
-            Some(make_populated_accountant_config_with_defaults());
-        let payable_dao_factory = Box::new(PayableDaoFactoryMock::new(PayableDaoMock::new()));
-        let receivable_dao_factory =
-            Box::new(ReceivableDaoFactoryMock::new(ReceivableDaoMock::new()));
-        let pending_payable_dao_factory = Box::new(PendingPayableDaoFactoryMock::new(
-            PendingPayableDaoMock::default(),
-        ));
-        let banned_dao_factory = Box::new(BannedDaoFactoryMock::new(BannedDaoMock::new()));
+        // TODO: Verify Scanners are defaulted properly [write this test once GH-574's recorder's code is merged, or cherry-pick the commit]
```

Comment:
There is a test for periodical scanning, do you think it is necessary to work on this todo.

Reply:
We have tests for periodical scanning, this kind of test looks like too much.